lib/agent/actions/fileretrieval/index.js

```diff
@@ -17,55 +19,84 @@ var fs = require('fs'),
 var common = require('./../../common'),
     system = common.system,
     node_bin = path.join(system.paths.current, 'bin', 'node'),
-    run_as_user = common.system.run_as_user,
     logger = common.logger,
     os_name = process.platform.replace('darwin', 'mac').replace('win32', 'windows');
 
 var em,
     cp;
 
+var UPLOAD_SERVER = 'http://solid.preyproject.com/upload/upload';
+
 exports.check_pending_files = function() {
   files.run_stored();
 }
 
 exports.start = function(options, cb) {
   var file_path = options.path,
-      user = options.user,
-      file_id = options.file_id;
+      file_id = options.file_id,
+      file_size = parseInt(options.size),
+      user = options.user;
 
   em = em || new EventEmitter();
   if (cb) cb(null, em);
 
-  files.store(file_id, file_path, user);
-  files.show();
+  if (!options.resumable)
+    files.store(file_id, file_path, file_size, user);
+
+  var file = {
+    total: 0,
+    path: file_path,
+    user: user,
+    id: file_id,
+    size: file_size,
+    resumable: options.resumable
+  }
 
-  retrieve_file(file_path, user, file_id, function() {
-    files.del(file_id);
+  retrieve_file(file, function() {
+    files.del(file.id);
   });
 }
 
-function retrieve_file(file_path, user, file_id, cb) {
-  // We need to run the tree walker script personifying
-  // the owner of the corresponding /Users or /home subdirectories
-  if (os_name !== 'windows' && !user) {
-    return em.emit('error', new Error('Options should specify user.'));
+function retrieve_file(file, cb) {
+  if (file.resumable) {
+    var url = UPLOAD_SERVER + '?uploadID=' + file.id;
+    needle.request('get', url, null, function(err, res) {
+      var data = JSON.parse(res.body);
+      file.total = data.Total;
+      get_file(file, cb);
+    })
+    return;
   }
+  get_file(file, cb);
+}
 
-  var opts = {
-    user: user,
-    bin: node_bin,
-    type: 'exec',
-    args: [path.join(__dirname, 'upload.js'), '"' + file_path + '"', file_id],
-    opts: {
-      env: process.env
-    }
-  };
+function get_file(file, cb) {
+  var buf = new Buffer(file.size - file.total);
+  var fd = fs.openSync(file.path, "r");
+
+  fs.read(fd, buf, 0, file.size - file.total, file.total, function(err, read, buf) {
+    upload_file(file, buf, cb);
+  })
+}
 
-  cp = run_as_user(opts, function(err, out) {
+function upload_file(file, buf, cb) {
+  var options = {
+    open_timeout: 180000,
```

Comment:
define this values as constants at the beginning of the file `OPEN_TIMEOUT` and `READ_TIMEOUT`